src/Build/BackEnd/Components/Caching/ResultsCache.cs

```diff
@@ -17,6 +18,11 @@ namespace Microsoft.Build.BackEnd
     /// </summary>
     internal class ResultsCache : IResultsCache
     {
+        /// <summary>
+        /// The presence of any of these flags affects build result for the specified request.
+        /// </summary>
+        private const BuildRequestDataFlags FlagsAffectingBuildResults = BuildRequestDataFlags.ProvideProjectStateAfterBuild;
```

Comment:
`SkipNonexistentTargets`
`IgnoreMissingEmptyAndInvalidImports`
`FailOnUnresolvedSdk`

also look like flags affecting the build result. Since the check is now behind a changewave, giving us a bit of extra safety, I would consider adding them here.